src/view/MixerElementVolumeWindow.vala

```diff
@@ -0,0 +1,109 @@
+/*
+ * brickman -- Brick Manager for LEGO MINDSTORMS EV3/ev3dev
+ *
+ * Copyright (C) 2016 Kaelin Laundry <wasabifan@outlook.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * MixerElementVolumeWindow.vala - Allows control of mixer element volume
+ */
+
+using Ev3devKit;
+using Ev3devKit.Ui;
+
+namespace BrickManager {
+    public class MixerElementVolumeWindow : BrickManagerWindow {
+        public signal void volume_up ();
+        public signal void volume_down ();
+        public signal void volume_min ();
+
+        IMixerElementViewModel _current_element = null;
+        bool _show_element_details = true;
+
+        Ui.Label element_label;
+
+        public MixerElementVolumeWindow () {
+            title = "Volume control";
+
+            element_label = new Ui.Label();
+            content_vbox.add(element_label);
+
+            var controls_menu = new Ui.Menu();
+
+            var volume_up_item = new Ui.MenuItem("+ Volume up");
+            volume_up_item.button.pressed.connect(() => {
+                if(_current_element != null)
+                    volume_up();
+            });
+            controls_menu.add_menu_item(volume_up_item);
+
+            var volume_down_item = new Ui.MenuItem("- Volume down");
+            volume_down_item.button.pressed.connect(() => {
+                if(_current_element != null)
+                    volume_down();
+            });
+            controls_menu.add_menu_item(volume_down_item);
+
+            var volume_min_item = new Ui.MenuItem("Mute");
+            volume_min_item.button.pressed.connect(() => {
+                if(_current_element != null)
```

Comment:
Between when it is created and when it is given an element, it will be null. But regardless of whether it _should_ be null, I'd like to design these classes so they won't throw unexpected errors even in the face of user error.